driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java

```diff
@@ -304,29 +308,26 @@ public void executeAsync(final AsyncReadBinding binding, final SingleResultCallb
     /**
      * Gets an operation whose execution explains this operation.
      *
-     * @param explainVerbosity the explain verbosity
+     * @param verbosity the explain verbosity
      * @return a read operation that when executed will explain this operation
      */
-    public ReadOperation<BsonDocument> asExplainableOperation(final ExplainVerbosity explainVerbosity) {
-        return new AggregateExplainOperation(getNamespace(), getPipeline())
-               .allowDiskUse(getAllowDiskUse())
-               .maxTime(getMaxAwaitTime(TimeUnit.MILLISECONDS), TimeUnit.MILLISECONDS)
-               .hint(wrapped.getHint())
-               .retryReads(getRetryReads());
+    public <R> ReadOperation<R> asExplainableOperation(@Nullable final ExplainVerbosity verbosity, final Decoder<R> resultDecoder) {
+        return new CommandReadOperation<R>(getNamespace().getDatabaseName(),
```

Comment:
Following the pattern used in FindOperation to just use a CommandReadOperation here instead of a custom class